src/ripple/rpc/handlers/HealthCheck.cpp

```diff
@@ -0,0 +1,119 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2014 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/misc/NetworkOPs.h>
+#include <ripple/json/json_value.h>
+#include <ripple/net/RPCErr.h>
+#include <ripple/protocol/jss.h>
+#include <ripple/rpc/Context.h>
+#include <ripple/rpc/impl/TransactionSign.h>
+#include <ripple/rpc/Role.h>
+
+namespace ripple {
+
+Json::Value
+doHealthCheck(RPC::JsonContext& context)
+{
+    bool constexpr humanReadable = true;
+    bool constexpr noAdmin = false;
+    bool constexpr noCounters = false;
+    auto info = context.netOps.getServerInfo(
+                    humanReadable, noAdmin, noCounters);
+
+    int last_validated_ledger_age = std::numeric_limits<int>::max();
+    if (info.isMember("validated_ledger"))
+        last_validated_ledger_age = info["validated_ledger"]["age"].asInt();
+    bool amendment_blocked = false;
+    if (info.isMember("amendment_blocked"))
+        amendment_blocked = true;
+    int number_peers = info["peers"].asInt();
+    std::string server_state = info["server_state"].asString();
+    auto load_factor = info["load_factor"].asDouble();
+
+    Json::Value ret = Json::objectValue;
+    enum {healthy, warning, critical};
+    int health = healthy;
+    auto set_health = [&health](int state)
+        {
+            if (health < state)
+                health = state;
+        };
+
+    if (last_validated_ledger_age >= 7)
+    {
+        ret[jss::info]["validated_ledger"] = last_validated_ledger_age;
+        if (last_validated_ledger_age < 20)
+            set_health(warning);
+        else
+            set_health(critical);
+    }
+
+    if (amendment_blocked)
+    {
+        ret[jss::info]["amendment_blocked"] = true;
+        set_health(critical);
+    }
+
+    if (number_peers <= 7)
+    {
+        ret[jss::info]["peers"] = number_peers;
```

Comment:
I kind of think we should populate "info" with the data even when that data is healthy. Meaning always return the number of peers. Same for the rest of these metrics.

Reply:
I thought about that, and we can do that.  But it makes it more difficult to know what is wrong when the status isn't Healthy.

Reply:
I haven't actually played with the tool, but my intuition aligns with @HowardHinnant's.  I think we want to present as little information as possible to most users.  If it's Healthy that's really all they want to know.  But, since there are folks like @cjcobb23 who always want all the information, we might consider a `--verbose` argument?

Reply:
We kind of already have it, but it is called `server_info`.  All of this information is collected from that command, and this command just filters that and spits it back out with attitude.

Reply:
@HowardHinnant I guess you are right, that all of that information is in `server_info`. I'm fine then with leaving this command as you have implemented it here.